### DISCUSSION TITLE
Entity manager not removing inactive entities?

### DISCUSSION THREAD

**Discussion by Lzrs111** [excaliburjs/Excalibur#2715] [Q&A] [answered]
Hi,

I'm making a game which is basically a Vampire Survivors clone - so lots of enemies, projectiles and Actors in general. I've noticed that after a while on mobile especially the game just kinda slows down and crashes. My guess is there's a memory leak somewhere. After checking the code on my side, I can't find any problems, so I decided to take heap snapshots and it seems the Entity Manager _entititiesToRemove array does not get cleared, just gets filled with more entities:

Snapshot 1:
![image](https://github.com/excaliburjs/Excalibur/assets/19205191/68e8c7c1-db75-48dc-93d0-73f729ce65c5)

Snapshot2 :
![image](https://github.com/excaliburjs/Excalibur/assets/19205191/d94e60c1-1ad6-46c3-9ba0-0e7f25b56658)

I think this might be the source of the problem. Can you please check and confirm? Thanks

**Comment by eonarheim** [comment, marked as answer]
@Lzrs111 Yup I think you spotted a memory leak! I can confirm this behavior locally.

I'm looking at the code and I think I've spotted the issue.

<img width="356" alt="image" src="https://github.com/excaliburjs/Excalibur/assets/612071/5bc50f41-19a2-4237-9671-cf24625ab17f">

**Reply by eonarheim** [reply]
The fix should be available in the alpha bits soon https://www.npmjs.com/package/excalibur?activeTab=versions

https://github.com/excaliburjs/Excalibur/pull/2727

**Reply by Lzrs111** [reply]
Hi Erik, 

Good news! Hope this fixes things on my end as well. 

Thanks :)
